### PR TITLE
Remove specs pages from docs

### PR DESCRIPTION
**Pull request by dstansby**
These are just empty pages now, so I've replaced them with a link to the zarr specs on the home page. This cleans up the top level table of contents a little bit.

### CODE REVIEW

**Comment by d-v-b** [comment]
+1 for removing pages that are currently empty, but I think we will need a place in the docs to publish specs for things like consolidated metadata, dtypes, and codecs until the logjam over in `zarr-specs` clears up.

**Comment by dstansby** [comment]
> we will need a place in the docs to publish specs

Do you think it's worth adding an empty page somewhere in this PR, or adding it later alongside the actual content?

**Comment by d-v-b** [comment]
lets wait for when we have actual content

**Comment by joshmoore** [comment]
I'm typically anti breaking any URLs. I'd suggest we redirect the spec/vX pages to the new homes and keep the top-level page also with the links to the new homes.

**Comment by dstansby** [comment]
I've redirected the deleted pages. I'm +1 to getting rid of all these pages and cleaning the top level navigation up a bit, and not sure there's much point keeping the top-level page given it just links out without any of it's own content.

**Review by normanrz — APPROVED** [review]

File: docs/index.rst [hunk 1]
```diff
@@ -27,6 +26,7 @@ Zarr-Python
 `Source Repository <https://github.com/zarr-developers/zarr-python>`_ |
 `Issue Tracker <https://github.com/zarr-developers/zarr-python/issues>`_ |
 `Zulip Chat <https://ossci.zulipchat.com/>`_
```
Comment (joshmoore) [inline docs/index.rst:28]:
```suggestion
`Zulip Chat <https://ossci.zulipchat.com/>`_ |
```

File: docs/conf.py [hunk 2]
```diff
@@ -81,6 +82,13 @@
 version = get_version("zarr")
 release = get_version("zarr")
 
+redirects = {
+     "docs/spec": "https://zarr-specs.readthedocs.io",
```
Comment (joshmoore) [inline docs/conf.py:86]: I'm struggling to test these under https://zarr--2555.org.readthedocs.build/en/2555/

https://zarr--2555.org.readthedocs.build/en/2555/spec/index.html e.g. is 404

**Review by joshmoore — CHANGES_REQUESTED** [review]
One minor typo and some issues with the redirects

File: docs/conf.py [hunk 3]
```diff
@@ -81,6 +82,13 @@
...
+     "docs/spec/v1": 'https://zarr-specs.readthedocs.io/en/latest/v1/v1.0.html',
+     "docs/spec/v2": "https://zarr-specs.readthedocs.io/en/latest/v2/v2.0.html",
+     "docs/spec/v3": "https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html",
```
Comment (dstansby) [inline docs/conf.py:89]:
```suggestion
     "spec": "https://zarr-specs.readthedocs.io",
     "spec/v1": 'https://zarr-specs.readthedocs.io/en/latest/v1/v1.0.html',
     "spec/v2": "https://zarr-specs.readthedocs.io/en/latest/v2/v2.0.html",
     "spec/v3": "https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html",
```

File: docs/conf.py (hunk 2 above)
Comment (dstansby) [reply docs/conf.py]: Should be working now - the redirect is from https://zarr--2555.org.readthedocs.build/en/2555/spec.html (corresponding to https://zarr.readthedocs.io/en/stable/spec.html).

File: docs/conf.py [hunk 4]
```diff
@@ -81,6 +82,13 @@
 version = get_version("zarr")
 release = get_version("zarr")
 
+redirects = {
+     "spec": "https://zarr-specs.readthedocs.io",
```
Comment (joshmoore) [inline docs/conf.py:86]: still not working for:
https://zarr--2555.org.readthedocs.build/en/2555/spec
https://zarr--2555.org.readthedocs.build/en/2555/spec/
https://zarr--2555.org.readthedocs.build/en/2555/spec/index.html
https://zarr--2555.org.readthedocs.build/en/2555/spec/v1/
https://zarr--2555.org.readthedocs.build/en/2555/spec/v1/index.html

Comment (dstansby) [reply docs/conf.py:86]: None of those are correct links for the current pages before this PR. The correct ones (that redirect for me) are:

- https://zarr--2555.org.readthedocs.build/en/2555/spec.html
- https://zarr--2555.org.readthedocs.build/en/2555/spec/v1.html
- https://zarr--2555.org.readthedocs.build/en/2555/spec/v2.html
- https://zarr--2555.org.readthedocs.build/en/2555/spec/v3.html

Comment (joshmoore) [reply docs/conf.py:86]: well that would explain it 🤦🏽

**Review by joshmoore — APPROVED** [review]

**Pull request merged**